src/OSPSuite.Core/Domain/Builder/ModelConfiguration.cs

```diff
@@ -1,28 +1,37 @@
-﻿namespace OSPSuite.Core.Domain.Builder
+﻿using OSPSuite.Core.Domain.Services;
+
+namespace OSPSuite.Core.Domain.Builder
 {
-   internal class ModelConfiguration
+   public class ModelConfiguration
    {
       public SimulationBuilder SimulationBuilder { get; }
       public IModel Model { get; }
       public SimulationConfiguration SimulationConfiguration { get; }
+      public ReplacementContext ReplacementContext { get; private set; }
```

Comment:
Introduce the replacement context that is caching what needs to be replaced instead of always computing it